tools/csv-merger/csv-merger.go

```diff
@@ -0,0 +1,375 @@
+			*specDescription,
+			*operatorImage,
+			replaces,
+			version,
+			*crdDisplay,
+		)
+		csvExtended := ClusterServiceVersionExtended{
+			TypeMeta:   csvBase.TypeMeta,
+			ObjectMeta: csvBase.ObjectMeta,
+			Spec:       ClusterServiceVersionSpecExtended{ClusterServiceVersionSpec: csvBase.Spec},
+			Status:     csvBase.Status}
+
+		// This is the base deployment + rbac for the OpenStack Cluster CSV
+		installStrategyBase := operator.GetInstallStrategyBase(
+			*namespace,
+			*operatorImage,
+			"IfNotPresent",
+		)
+
+		for _, image := range strings.Split(*relatedImagesList, ",") {
+			if image != "" {
+				name := ""
+				if strings.Contains(image, "|") {
+					image_s := strings.Split(image, "|")
+					image = image_s[0]
+					name = image_s[1]
+				} else {
+					names := strings.Split(strings.Split(image, "@")[0], "/")
+					name = names[len(names)-1]
+				}
+				csvExtended.Spec.RelatedImages = append(
+					csvExtended.Spec.RelatedImages,
+					relatedImage{
+						Name: name,
+						Ref:  image,
+					})
+			}
+		}
+
+		for _, csvStr := range csvs {
+			if csvStr != "" {
+				csvBytes := []byte(csvStr)
+
+				csvStruct := &csvv1alpha1.ClusterServiceVersion{}
+
+				err := yaml.Unmarshal(csvBytes, csvStruct)
+				if err != nil {
+					panic(err)
+				}
+
+				installStrategyBase.DeploymentSpecs = append(installStrategyBase.DeploymentSpecs, csvStruct.Spec.InstallStrategy.StrategySpec.DeploymentSpecs...)
+				installStrategyBase.ClusterPermissions = append(installStrategyBase.ClusterPermissions, csvStruct.Spec.InstallStrategy.StrategySpec.ClusterPermissions...)
+				installStrategyBase.Permissions = append(installStrategyBase.Permissions, csvStruct.Spec.InstallStrategy.StrategySpec.Permissions...)
+
+				for _, owned := range csvStruct.Spec.CustomResourceDefinitions.Owned {
+					csvExtended.Spec.CustomResourceDefinitions.Owned = append(
+						csvExtended.Spec.CustomResourceDefinitions.Owned,
+						csvv1alpha1.CRDDescription{
+							Name:        owned.Name,
+							Version:     owned.Version,
+							Kind:        owned.Kind,
+							Description: owned.Description,
+							DisplayName: owned.DisplayName,
+						},
+					)
+				}
+
+				csv_base_alm_string := csvExtended.Annotations["alm-examples"]
+				csv_struct_alm_string := csvStruct.Annotations["alm-examples"]
+				var base_almcrs []interface{}
+				var struct_almcrs []interface{}
+				if err = json.Unmarshal([]byte(csv_base_alm_string), &base_almcrs); err != nil {
+					panic(err)
+				}
+				if err = json.Unmarshal([]byte(csv_struct_alm_string), &struct_almcrs); err == nil {
+					//panic(err)
+					for _, cr := range struct_almcrs {
+						base_almcrs = append(
+							base_almcrs,
+							cr,
+						)
+					}
+				}
+				alm_b, err := json.Marshal(base_almcrs)
+				if err != nil {
+					panic(err)
+				}
+				csvExtended.Annotations["alm-examples"] = string(alm_b)
+
+			}
+		}
+
+		hidden_crds := []string{}
+		visible_crds := strings.Split(*visibleCRDList, ",")
```

Comment:
We don't appear to be passing these in, yet, so presumably everything is visible in OLM?

Reply:
I think we will start using these soon to suppress the visiblity of CRDs in the OpenShift UI/console. NOTE: this setting has no bearing on the visibility of the crds on the command line or API.